File: desktop/src/com/mygdx/game/desktop/DesktopLauncher.java
package com.mygdx.game.desktop;

import com.badlogic.gdx.backends.lwjgl.LwjglApplication;
import com.badlogic.gdx.backends.lwjgl.LwjglApplicationConfiguration;
import java.game.com.mygdx.NotPhysicsEngineMain;

public class DesktopLauncher {
	public static void main (String[] arg) {
		LwjglApplicationConfiguration config = new LwjglApplicationConfiguration();
		config.width=1600;
		config.height=900;
		//config.fullscreen=true;
		new LwjglApplication(new NotPhysicsEngineMain(), config);
	}
}

File: desktop/src/main/scala/npe/desktop/DesktopLauncher.scala
package npe.desktop

import com.badlogic.gdx.backends.lwjgl.{LwjglApplication, LwjglApplicationConfiguration}
import npe.game.{NotPhysicsEngineCLI, NotPhysicsEngineGUI}
import npe.tools.JsonReader
import org.json.*
import org.slf4j.LoggerFactory

object DesktopLauncher {
    def main(args: Array[String]): Unit = {
        val logger = LoggerFactory.getLogger(classOf[DesktopLauncher])
        val interface_type = args.apply(0)
        logger.info("interface_type:" + interface_type)
        if (interface_type.equals("cli")) {
            // cli界面
            val aaa = new NotPhysicsEngineCLI
//            aaa.npec()
        }
        else if (interface_type.equals("gui")) {
            // gui界面
            val config = new LwjglApplicationConfiguration
            val settings_json = JsonReader.read_json_from_path("json_files/settings.json")
            val desktop_setting = settings_json.getJSONObject("desktop_config")
            config.width = desktop_setting.getInt("width");
            config.height = desktop_setting.getInt("height");
            new LwjglApplication(new NotPhysicsEngineGUI, config)
        }

    }
}

class DesktopLauncher {
    
}